packages/web/stores/assets/assets-store.ts

```diff
@@ -157,49 +153,53 @@ export class ObservableAssets {
     sourceChainNameOverride?: string;
   })[] {
     return this.assets
-      .filter((asset) => asset.origin_chain_id !== this.chain.chainId) // Filter osmosis native assets
+      .filter((asset) => asset.transferMethods.length > 0) // Filter osmosis native assets
       .filter((asset) => {
         if (typeof window === "undefined") return true;
 
         if (sessionStorage.getItem(UnlistedAssetsKey) === "true") {
           return true;
         }
-        return !asset.keywords?.includes("osmosis-unlisted");
+        return !asset.preview;
       }) // Remove unlisted assets if preview assets is disabled
       .map((ibcAsset) => {
-        const chainInfo = this.chainStore.getChain(ibcAsset.origin_chain_id);
+        const chainInfo = this.chainStore.getChain(
+          ibcAsset.counterparty[0]?.chainId?.toString() ?? ""
+        );
 
-        const minimalDenom = getSourceDenomFromAssetList(ibcAsset);
+        const sourceDenom_ = ibcAsset.sourceDenom;
         const originCurrency = chainInfo.currencies.find((cur) => {
-          if (typeof minimalDenom === "undefined") return false;
+          if (typeof sourceDenom_ === "undefined") return false;
 
-          if (minimalDenom.startsWith("cw20:")) {
+          if (sourceDenom_.startsWith("cw20:")) {
             /** Note: since we're searching on counterparty config, the coinMinimalDenom
              *  is not the Osmosis IBC denom, it's the source denom
              */
-            return cur.coinMinimalDenom.startsWith(minimalDenom);
+            return cur.coinMinimalDenom.startsWith(sourceDenom_);
           }
-          return cur.coinMinimalDenom === minimalDenom;
+          return cur.coinMinimalDenom === sourceDenom_;
         });
 
         if (!originCurrency) {
           throw new Error(
-            `Unknown currency ${minimalDenom} for ${ibcAsset.origin_chain_id}`
+            `Unknown currency ${sourceDenom_} for ${
+              ibcAsset.counterparty![0].chainId
+            }`
           );
         }
 
-        const ibcTrace = getLastIbcTrace(ibcAsset.traces);
+        const ibcTransferMethod = ibcAsset.transferMethods.find(
+          ({ type }) => type === "ibc"
+        ) as IbcTransferMethod | undefined;
```

Comment:
Perhaps we can create a utility for this? I say this because of the type. Having to constantly cast it as IbcTransferMethod could become a hassle.

Reply:
Typically I'd agree that repeated code is bad, but IMO this is more readable than a utility that saves a few LOC because we can see exactly what is going on. I noticed with the many utilities for assets that we had (some I removed) made asset-related code hard to read because I often found myself diving into the implementation to see what was going on.